### ISSUE TITLE
Helper loading problem: package vs. framework

### ISSUE THREAD

**Issue by leonardor** [bcit-ci/CodeIgniter#2045]
Hello!

I have written a package that improves some of the framework helpers with new functions and they are named alike. I am aware that the extended helpers should be named with the 'my' prefix, but I observed that the precedence of loading the helpers is working fine, even if the package helper is named exactly the same as the framework helper. The version of the CodeIgniter framework I work with is 2.1.0.

So, the problem is:

1) When loading a helper (for example the 'url_helper') of the custom package from the autoload file like this:

autoload.php:

``` php
//...
$autoload['packages'] = array(BASEPATH.'third_party/custom_package');
$autoload['helper'] = array('url');
//...
```

 the 'url_helper' is loaded fine - as I described.

2) If the 'url_helper' is loaded via another helper like this:

autoload.php:

``` php
//...
$autoload['packages'] = array(BASEPATH.'third_party/custom_package');
$autoload['helper'] = array();
//...
```

custom_helper.php:

``` php
//...
$CI =& get_instance();
$CI->load->helper('url');
//...
```

then the precedence is messed up so the helper is loaded from the framework and not from the package.

In conclusion, either the autoload way of loading is wrong or the instance way of loading is buggy.
